### DISCUSSION TITLE
Property '[$type]' is missing in type 'IModelType...'

### DISCUSSION THREAD

**Discussion by shubham-rateria** [mobxjs/mobx-state-tree#1832] [Q&A]
Hey,

I am trying to create a separate package to hold my MST models and reuse across projects. These packages are written in TypeScript. When I build this package and try to reuse it in my front-end code (React, CRA + TypeScript), I get a ton of type errors, most common of which being -

```
Property '[$type]' is missing in type 'IModelType<{ refid: ISimpleType<string>; name: IOptionalIType<ISimpleType<string>, [undefined]>; size: IOptionalIType<ISimpleType<number>, [...]>; ... 25 more ...; loading: IOptionalIType<...>; }, { ...; } & ... 1 more ... & { ...; }, _NotCustomized, _NotCustomized>' but required in type 'IType<any, any, object>'
```

Can you help solve this?

Versions -

```
"mobx": "^6.3.7",
"mobx-react": "^7.2.1",
"mobx-state-tree": "^5.0.1"
```

Please let me know what additional information I can provide. Thanks for your help!

**Comment by Arasid** [comment]
Hi @shubham-rateria, I don't know if you were able to fix it, but if not, the solution that worked for me is in issue #1398.
Just get the `mobx-state-tree` versions to match across projects.
